### PR TITLE
Adds prime-factors exercise

### PR DESCRIPTION
**Pull request by amscotti**
Hi, this PR is to add a new exercise for `prime-factors`, [here is a link to the problem-specifications](https://github.com/exercism/problem-specifications/tree/master/exercises/prime-factors)

I'm still learning Dart, so feel free to point out anything that needs to be change or improved. I'm also happy to change the `config.json` to better fit your layout for the track.

Thanks! :)

### CODE REVIEW

File: exercises/prime-factors/test/prime_factors_test.dart [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import 'package:prime_factors/prime_factors.dart';
+import 'package:test/test.dart';
+
+void main() {
+  final primeFactors = new PrimeFactors();
+
+  group('PrimeFactors', () {
+    group('returns prime factors for the given input number', () {
+      test("no factors", () {
+        final List<Object> result = primeFactors.factors(1);
```
Comment (Stargator) [inline exercises/prime-factors/test/prime_factors_test.dart]: The return type should be `List<int>`

**Comment by amscotti** [comment]
@Stargator Updated, are there any other changes you would like?

File: config.json [hunk 2]
```diff
@@ -348,6 +348,21 @@
         "loops",
         "math"
       ]
+    },
+    {
+      "slug": "prime-factors",
+      "uuid": "89bb2ba8-86f2-4660-8739-4141eb9e6cf5",
+      "core": false,
+      "unlocked_by": null,
+      "difficulty": 5,
```
Comment (Stargator) [inline config.json]: I would recommend a difficulty of 3

**Comment by Stargator** [comment]
@amscotti I've been thinking about where to place the exercise in the order. I would recommend making it after anagram. But I don't think it has to be a core exercise.

So the needed change is just to set unlocked by `anagram`. If possible, also move it to after anagram in the list.

That should be it.

**Comment by amscotti** [comment]
@Stargator This should be all set, I make it unlocked by `anagram` and moved it up the list to be next to `anagram`. Let me know if you see any other issues you want changed.

Thanks!

**Pull request merged**